### PR TITLE
Add support for nested maps to has-key

### PR DESCRIPTION
**Pull request by jathak**
See https://github.com/sass/sass-spec/pull/1558

### CODE REVIEW

File: lib/src/functions/map.dart [hunk 1]
```diff
@@ -67,10 +67,19 @@ final _values = _function(
...
         arguments[0].assertMap("map").contents.values, ListSeparator.comma));
 
-final _hasKey = _function("has-key", r"$map, $key", (arguments) {
+final _hasKey = _function("has-key", r"$map, $key, $keys...", (arguments) {
   var map = arguments[0].assertMap("map");
-  var key = arguments[1];
-  return SassBoolean(map.contents.containsKey(key));
+  var allKeys = [arguments[1], ...arguments[2].asList];
+  var intermediateKeys = allKeys.sublist(0, allKeys.length - 1);
+  for (var key in intermediateKeys) {
```
Comment (nex3) [inline lib/src/functions/map.dart:74]:
```suggestion
  for (var key in allKeys.take(allKeys.length - 1)) {
```

Not only is this terser, it also avoids creating an unnecessary intermediate list :smiley:.

**Review by nex3 — CHANGES_REQUESTED** [review]
Please add a CHANGELOG entry as well.

File: lib/src/functions/map.dart (hunk 1 above)
Comment (jathak) [reply lib/src/functions/map.dart]: Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 1.27.0
+
+* Add support for nested maps in `map.has-key`.
```
Comment (Awjin) [inline CHANGELOG.md:3]: This should also link to the docs.

Comment (jathak) [reply CHANGELOG.md:3]: Will do. I'm going to wait until #1076 is merged

[the remark this answers is not in the document]
Comment (jathak) [reply CHANGELOG.md:9]: Done

**Review by nex3 — APPROVED** [review]

**Pull request merged**